server/src/main/java/io/crate/execution/dml/Indexer.java

```diff
@@ -370,10 +370,7 @@ public void verify(Object providedValue) {
             Object generatedValue = input.value();
             int compare = Comparator
                 .nullsFirst(valueType)
-                .compare(
-                    valueType.sanitizeValue(generatedValue),
-                    valueType.sanitizeValue(providedValue)
-                );
+                .compare(valueType.sanitizeValue(generatedValue), providedValue);
```

Comment:
Could `providedValue` be not sanitized when `verify()` is invoked by `ObjectIndexer` and the `innerValue` is a synthetic?

Reply:
Good point. I'll revert this part